Source/Core/Core/HW/EXI/EXI_DeviceIPL.cpp

```diff
@@ -347,68 +290,88 @@ void CEXIIPL::TransferByte(u8& _uByte)
       {
         // "Queue Length"... return 0 cause we're instant
-        _uByte = 0;
+        *data = 0;
       }
-      break;
-
-    case REGION_EUART_UNK:
-      // Writes 0xf2 then 0xf3 on EUART init. Just need to return non-zero
-      // so we can leave the byte untouched.
-      break;
-
-    case REGION_UART_UNK:
-      DEBUG_LOG(OSREPORT, "UART? %x", _uByte);
-      _uByte = 0xff;
-      break;
-
-    case REGION_BARNACLE:
-      DEBUG_LOG(OSREPORT, "UART Barnacle %x", _uByte);
-      break;
-
-    case REGION_WRTC0:
-    case REGION_WRTC1:
-    case REGION_WRTC2:
-    // Wii only RTC flags... afaik just the Wii Menu initialize it
-    default:
-      if ((m_uAddress >> 6) < ROM_SIZE)
-      {
-        if (!IsWriteCommand())
-        {
-          u32 position = ((m_uAddress >> 6) & ROM_MASK) + m_uRWOffset;
+    };
 
-          // Technically we should descramble here iff descrambling logic is enabled.
-          // At the moment, we pre-decrypt the whole thing and
-          // ignore the "enabled" bit - see CEXIIPL::CEXIIPL
-          _uByte = m_pIPL[position];
+    if (IN_RANGE(ROM))
+    {
+      if (!m_command.is_write())
+      {
+        u32 dev_addr = DEV_ADDR_CURSOR(ROM);
+        // Technically we should descramble here iff descrambling logic is enabled.
+        // At the moment, we pre-decrypt the whole thing and
+        // ignore the "enabled" bit - see CEXIIPL::CEXIIPL
+        data = m_rom[dev_addr];
 
-          if ((position >= 0x001AFF00) && (position <= 0x001FF474) && !m_FontsLoaded)
+        if ((dev_addr >= 0x001AFF00) && (dev_addr <= 0x001FF474) && !m_fonts_loaded)
+        {
+          if (dev_addr >= 0x001FCF00)
           {
-            if (position >= 0x001FCF00)
-            {
-              PanicAlertT("Error: Trying to access Windows-1252 fonts but they are not loaded. "
-                          "Games may not show fonts correctly, or crash.");
-            }
-            else
-            {
-              PanicAlertT("Error: Trying to access Shift JIS fonts but they are not loaded. "
-                          "Games may not show fonts correctly, or crash.");
-            }
-            m_FontsLoaded = true;  // Don't be a nag :p
+            PanicAlertT("Error: Trying to access Windows-1252 fonts but they are not loaded. "
+                        "Games may not show fonts correctly, or crash.");
           }
+          else
+          {
+            PanicAlertT("Error: Trying to access Shift JIS fonts but they are not loaded. "
+                        "Games may not show fonts correctly, or crash.");
+          }
+          // Don't be a nag
+          m_fonts_loaded = true;
         }
       }
+    }
+    else if (IN_RANGE(SRAM))
+    {
+      u32 dev_addr = DEV_ADDR_CURSOR(SRAM);
+      if (m_command.is_write())
+        g_SRAM.raw[dev_addr] = data;
       else
+        data = g_SRAM.raw[dev_addr];
+    }
+    else if (IN_RANGE(UART))
+    {
+      switch (DEV_ADDR(UART))
+      {
+      case 0:
```

Comment:
Shouldn't constants be defined for these UART magics.

Reply:
they are just unknown register indices